### DISCUSSION TITLE
Model containing a variable number of sub-models

### DISCUSSION THREAD

**Discussion by antoine-farel** [fastapi/fastapi#6374] [Questions] [answered]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel, Field

class Person(BaseModel):
   age: int = Field(..., example=10)
   length : float = Field(..., example=1.76)
   hairs: str = Field(..., example='brown')

class PeopleNames(BaseModel):
   names: list[str] = Field(..., example=['Basile', 'John'])

app = FastAPI()

@app.post("/people", summary = "Get info about people")
def get_people_infos(people : PeopleNames) -> dict:
   data = {}
   for name in people['names']:
      data[name] = get_infos(name) #with get_infos the function to retrieve the information for the given name
   return data
```

### Description

I want to add a response_model to my function that respects what my function can return.

### Operating System

Windows

### FastAPI Version

0.75.0

### Python Version

3.9.12

### Additional Context

I thought of a model like this:
```python
class Peoples(BaseModel):
   data : dict[str, Person] = Field(..., example = {'John' : {'age' : 10, 'length' : 1.76, 'hairs' : 'brown'}, 'Basile' : {'age' : 20, 'length' : 1.82, 'hairs' : 'blond'}})
```
but a dictionary containing only the data key is necessary for it to work, which I don't want.
How do I achieve what I want to do?
Thank you!

**Comment by antoine-farel** [comment, marked as answer]
> Why does person have no name?
> 
> Person should have a name (or some unique identifier).
> 
> ```python
> class Person(BaseModel):
>    name: str = Field(..., example='John')
>    age: int = Field(..., example=10)
>    length : float = Field(..., example=1.76)
>    hairs: str = Field(..., example='brown')
> ```
> 
> Then your response should be something like
> 
> (this is untested but you get the idea)
> 
> ```python
> class GetPeopleResponse(BaseModel):
>    data : List[Person] = Field(..., example = [{'name':'John', 'age' : 10, 'length' : 1.76, 'hairs' : 'brown'}, {'name': 'Basile', 'age' : 20, 'length' : 1.82, 'hairs' : 'blond'}])
> ```
> 
> edit, or does example have to be a dict? 🤔

I need to make some changes in my inputs/outputs but indeed it seems to work well.
The solution is pretty simple but I was in tunnel vision, thanks for your help!

**Comment by YuriiMotov** [comment]
The following also works fine in current FastAPI version (0.115.12):

```python
from fastapi import FastAPI
from pydantic import BaseModel, Field


class Person(BaseModel):
   age: int = Field(..., example=10)

class PeopleNames(BaseModel):
   names: list[str] = Field(..., example=['Basile', 'John'])

app = FastAPI()

@app.post("/people", summary = "Get info about people")
def get_people_infos(people : PeopleNames) -> dict[str, Person]:
   data = {}
   for name in people.names:
      data[name] = Person(age=1)
   return data

```
